components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -3574,6 +3595,35 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_year_as_string() {
+        test_none_with_ctx_and_extra(cast_year_as_string);
+
+        let cs: Vec<(i64, Vec<u8>, String)> = vec![
```

Comment:
Please add more tests according to thttps://dev.mysql.com/doc/refman/8.0/en/year.html

Reply:
I think we don't need to match the behavior on the document here, I am migrating the function from TiDB to TiKV, and the only special case is for `0`, as for cases like 69 -> 2069, it is handled on insert, and will not use this function.